extensions/protobuf/src/main/java/org/modelmapper/protobuf/ProtobufHelper.java

```diff
@@ -57,6 +57,10 @@ public static boolean hasBuilder(Class<?> builderType, String field) {
     throw new Errors().addMessage("Invalid protocol buffer type: %s", type.getName()).toConfigurationException();
   }
 
+  public static Class<?> fieldType(Class<?> type, String memberName) throws NoSuchFieldException {
+    return type.getDeclaredField(formatFieldName(memberName)).getType();
+  }

```

Comment:
With `type.getDeclaredField(formatFieldName(memberName)).getType()` we receive `java.lang.Object` as type for `java.lang.String` types, therefore i would only use declared fields in case of Iterable types and fall back to the getter method's return type (as in earlier implementation) in any other case:

```suggestion
  public static Class<?> fieldType(Class<?> type, String memberName) throws NoSuchFieldException, NoSuchMethodException {
    final Class<?> fieldType =  type.getDeclaredField(formatFieldName(memberName)).getType();
    if (Iterable.class.isAssignableFrom(fieldType)) {
      return fieldType;
    } else {
      return getter(type, memberName).getReturnType();
    }
  }
```

Of course the `NoSuchMethodException` has to be handled in `ProtobufValueWriter`.